core-processor/src/executor.rs

```diff
@@ -24,102 +24,60 @@ use crate::{
     configs::ExecutionSettings,
     ext::ProcessorExt,
 };
-use alloc::{collections::BTreeMap, string::ToString, vec::Vec};
+use alloc::{
+    collections::{BTreeMap, BTreeSet},
+    string::ToString,
+    vec::Vec,
+};
 use gear_backend_common::{BackendReport, Environment, IntoExtInfo, TerminationReason};
 use gear_core::{
     env::Ext as EnvExt,
     gas::{ChargeResult, GasAllowanceCounter, GasCounter, ValueCounter},
-    memory::{AllocationsContext, WasmPageNumber},
+    ids::ProgramId,
+    memory::{AllocationsContext, Memory, PageBuf, PageNumber, WasmPageNumber},
     message::{ContextSettings, IncomingDispatch, MessageContext},
 };
 
```

Comment:
That's really strange, that refactoring increases amount of lines by 20%.
But ok, it seems ok for that moment, due to need of final refactoring of core-processor, once all api and use cases would be clear.

Reply:
That's really strange to evaluate refactoring by line numbers

Reply:
Just friendly notice.
I mean that in future some good trait for executor should be present